sdk/textanalytics/azure-ai-textanalytics/azure/ai/textanalytics/_text_analytics_client.py

```diff
@@ -544,20 +550,34 @@ def begin_cancel_analyze_healthcare(  # type: ignore
                 :caption: Cancel an existing health operation.
         """
         polling_interval = kwargs.pop("polling_interval", 5)
-        initial_response = getattr(poller._polling_method, "_initial_response") # pylint: disable=protected-access
-        operation_location = initial_response.http_response.headers["Operation-Location"]
 
-        job_id = urlparse(operation_location).path.split("/")[-1]
+        terminal_states = ["cancelled", "cancelling", "failed", "succeeded", "partiallyCompleted", "rejected"]
+        analyze_healthcare_entities_poller._thread.join() # Join so we no longer have to await the result of this poller
+        analyze_healthcare_entities_poller._polling_method.update_status()
+
+        if analyze_healthcare_entities_poller._polling_method.status() in terminal_states:
+            print("Operation with ID '%s' is already in a terminal state and cannot be cancelled." \
```

Comment:
should this be raised as an error? or does Python knows how to mask this print statement into something the user can see

Reply:
Maybe an error would be better.  It's a little bit difficult to test every code path here since most of the time the jobs succeed quickly enough that this always happens. I could try adding larger documents to the request to see if i can get the job to run long enough to test actual cancellation.

Reply:
I decided to raise the exception type `Warning` in the new cancellation method on the poller.  I thought this type made the most sense of all the built-in exceptions, but I can change it to something else if anyone has any suggestions.

Reply:
hmm idk what the behavior is in Python for this scenarios. @iscai-msft for context

Reply:
for python we can log warnings, usually they look like [this](https://github.com/Azure/autorest.python/blob/autorestv3/autorest/codegen/__init__.py#L152), not Warning raises. I think if other languages are raising an error when trying to cancel a finished operation, we might as well just raise an error too though

Reply:
I got the inspiration for this from [here](https://docs.python.org/3.7/library/exceptions.html#warnings).  If there is a better exception type to use, could you make a suggestion?